tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -111,7 +111,7 @@ __host__ __device__ inline double tf_max(double x, double y) {
 }
 
 // ROCM TODO re-enable them after adding fp16 support logic
-#if GOOGLE_CUDA
```

Comment:
@yxsamliu could you explain the builtin macros used by `hip-clang` toolchain?

generally speaking we want to avoid adding compiler-specific macros here.

Reply:
I think !__HCC___ may not be necessary here. I will try if I can remove it.

Reply:
@yxsamliu could you try switch to `TENSORFLOW_USE_ROCM` macro?

Reply:
@deven-amd / @yxsamliu this PR enables several shuffle operations for fp16 types.

And actually only `GpuShuffleXorSync` is used by kernels in `depthwise_conv_op_gpu.cu.cc` if I understand it correctly.